Adding changelog for v1.1.0-rc1 (#2967)
kubernetes-sigs/gateway-api: CHANGELOG.md

- [v0.1.0-rc2](#v010-rc2)
- [v0.1.0-rc1](#v010-rc1)

# v1.1.0-rc1

We expect that this release candidate is quite close to the final v1.1.0
release. However, breaking API changes are still possible.

This release candidate is suitable for implementors, but we do not recommend
shipping products based on a release candidate API due to the possibility of
incompatible changes prior to the final release. The following represents the
changes since v1.0.0:

## Standard Channel

### GRPCRoute has Graduated to GA 🎉

GRPCRoute has graduated to GA (v1) and is now part of the Standard Channel. If
you are already using the experimental version GRPCRoute, we recommend holding
off on upgrading to the standard channel version of GRPCRoute until the
controllers you're using have been updated to support GRPCRoute v1. Until then,
it is safe to upgrade to the experimental channel version of GRPCRoute in v1.1
that includes both v1alpha2 and v1 API versions.

> Leading Contributor: @gnossen

Author: @gnossen Any changes you'd like to make here?

Reviewer: Nope. This looks great! Thanks.

### Mesh Support

The standard for using Gateway API for Mesh has formally graduated to v1.0.

> Leading Contributors: @howardjohn, @keithmattix, @kflynn

Author: @howardjohn @keithmattix @kflynn can you help fill this out?

Reviewer 2: Yep - I want to make sure we add @mikemorris who's also been pushing this forward since the early days

Author: Great idea @keithmattix! Mind making a suggestion to formally make the change? @howardjohn just added some initial text here, feel free to update or revise that too.

### Conformance Profiles and Reports

The concept of conformance profiles and the corresponding reports have graduated
to GA.

> Leading Contributors: @mlavacca, @shaneutt

Author: @mlavacca @shaneutt can you help fill this out?

Reviewer 3: ```suggestion
The Conformance Reports API and the corresponding test suite have been graduated to GA. The Conformance report API has been expanded with the `mode` field (intended to specify the working mode of the implementation), and the `gatewayAPIChannel` (standard or experimental). The `gatewayAPIVersion` and `gatewayAPIChannel` are now filled in automatically by the suite machinery, along with a brief description of the testing outcome.
The Reports have been reorganized in a more structured way, and the implementations can now add information on how the tests have been run and provide reproduction steps.

Leading Contributors: @mlavacca, @shaneutt
```

### Port in ParentRefs

You can now use a `port` field in ParentRefs to attach to Gateways, Services, or
other parent resources.

> Leading Contributor: @frankbu

Author: @frankbu can you help fill this out?

## Experimental Channel

### Session Persistence + BackendLBPolicy
Session Persistence is being introduced to Gateway API via a new policy
(BackendLBPolicy) and Route-level configuration for both HTTPRoute and
GRPCRoute.

> Leading Contributors: @gcs278, @ginayeh

Author: @gcs278 @ginayeh can you help fill this out?

Reviewer 4: This seems reasonable as a high-level release note. Do you expect we should go into more detail? I don't have any in particular I think we should say, expect maybe what fields are being introduced (SessionName, Timeouts, Type...), but that might be too much for this release note based on the other examples I see.

Experimental PR has been created https://github.com/kubernetes-sigs/gateway-api/pull/2969.

Reviewer 4: Okay - I came up with a bit more detail, please tweak as you see fit. I can provide more detail if needed.

```suggestion
Session Persistence is being introduced to Gateway API via a new policy
(BackendLBPolicy) for Service-level configuration and as fields within HTTPRoute and
GRPCRoute for Route-level configuration. The BackendLBPolicy and Route-level
APIs provide the same session persistence configuration, including session
timeouts, session name, session type, and cookie lifetime type.

Leading Contributors: @gcs278, @ginayeh
```

### Gateway Client Cert Verification

> Gateways can now configure client cert verification for each Gateway Listener.

Reviewer 5: ```suggestion
Gateways can now configure client cert verification for each Gateway Listener by introducing a new field
`frontendValidation` field within `tls`. This field supports configuring a list of CA Certificates that can be used as a trust anchor to validate the certificates presented by the client.
```

> Leading Contributors: @arkodg

Author: @arkodg can you help fill this out?

### BackendTLSPolicy
As part of a broader goal of making our TLS terminology more consistent
throughout the API, we've introduced some breaking changes to BackendTLSPolicy.
This has resulted in a new API version (v1alpha3) and will require any existing
users of this policy to uninstall the v1alpha2 version before installing this
newer version.

> Leading Contributors: @candita

Author: @candita can you help fill this out?

### Gateway Params
Gateways now feature a new field that allows references to
implementation-specific parameters, similar to GatewayClass.

> Leading Contributors: @howardjohn

Author: @howardjohn can you help fill this out?

- [v0.1.0-rc2](#v010-rc2)
- [v0.1.0-rc1](#v010-rc1)

# v1.1.0-rc1

We expect that this release candidate is quite close to the final v1.1.0

> release. However, breaking API changes are still possible.

Reviewer 6: ```suggestion
release. However, subsequent breaking API changes are still possible.
```

This release candidate is suitable for implementors, but we do not recommend
shipping products based on a release candidate API due to the possibility of
incompatible changes prior to the final release. The following represents the
changes since v1.0.0:

## Standard Channel

### GRPCRoute has Graduated to GA 🎉

GRPCRoute has graduated to GA (v1) and is now part of the Standard Channel. If
you are already using the experimental version GRPCRoute, we recommend holding
off on upgrading to the standard channel version of GRPCRoute until the
controllers you're using have been updated to support GRPCRoute v1. Until then,
it is safe to upgrade to the experimental channel version of GRPCRoute in v1.1
that includes both v1alpha2 and v1 API versions.

Leading Contributor: @gnossen

> ### Mesh Support
> 
> The standard for using Gateway API for Mesh has formally graduated to v1.0.

Reviewer 7: ```suggestion
### Service Mesh Support has Graduated to GA 🎉

The standard for using Gateway API for Mesh has formally graduated to GA (v1) and is now part of the Standard Channel.

Service mesh support in Gateway API allows service mesh users to use the same API to manage ingress traffic and mesh traffic, reusing the same policy and routing interfaces. In Gateway API v1.1, routes (such as HTTPRoute) can now have a `Service` as a `parentRef`, to control how traffic to specific services behave. For more information, read the [service mesh](https://gateway-api.sigs.k8s.io/mesh/) documentation or the see the list of [integrations](https://gateway-api.sigs.k8s.io/implementations/#service-mesh-implementation-status).
```

Leading Contributors: @howardjohn, @keithmattix, @kflynn

### Conformance Profiles and Reports

The Conformance Reports API and the corresponding test suite have been graduated to GA. The Conformance report API has been expanded with the `mode` field (intended to specify the working mode of the implementation), and the `gatewayAPIChannel` (standard or experimental). The `gatewayAPIVersion` and `gatewayAPIChannel` are now filled in automatically by the suite machinery, along with a brief description of the testing outcome.

> The Reports have been reorganized in a more structured way, and the implementations can now add information on how the tests have been run and provide reproduction steps.

Reviewer 8: I just noticed you seem to be 80-lining for some lines above, but not these? NBD but if you feel like updating it :shrug:

Author: These are just the result of me accepting GitHub suggestions. Once they're all in, I'll update the line length for everything.

Reviewer 6: @robscott would you want to add anything about the work that @xtineskim completed?  It's user facing and it holds alot of meaning for some folks.

This has resulted in a new API version (v1alpha3) and will require any existing
users of this policy to uninstall the v1alpha2 version before installing this
newer version.

Leading Contributors: @candita

### Gateway Params
Gateways now feature a new field that allows references to
implementation-specific parameters, similar to GatewayClass.

Leading Contributors: @howardjohn

## Everything Else

### Validation Changes
- TLS Configuration is no longer required on Gateway Listeners to enable more
  flexible TLS configuration. (#2721, @robscott)

### Conformance

- Fixed GatewayWithAttachedRoutes conformance test to not check that the
  HTTPRoute status includes an "Accepted: False" condition because this is not
  required by the specification. (#2548, @frankbu)

### Cleanup

- The validating webhook has been removed. CEL validation is now built-in to
  CRDs and replaces the webhook. (#2595, @robscott)
- BackendTLSPolicy WellKnownCACerts field has been updated to
  implementation-specific support (#2741, @sunjayBhatia)
- Clarify policy attachment by two of the same policy types when using section
  names. (#2442, @maleck13)

> 

Reviewer 8: Am I blanking or did we have more GWCTL changes? :thinking: 

I was just thinking about this, since GWCTL is one of the more user-facing things we have... I swear I remember there being more stuff...

Author: @gauravkghildiyal is there anything we want to include in the changelog for gwctl?

Reviewer 9: ```suggestion
### gwctl

* We've extended the `get` command to support gateways, gatewayclasses, and
  namespaces. (#2865, #2782, #2847, @jongwooo)
* The `get` command now provides more detailed information for httproutes,
  policies, and policycrds. (#2805, #2808, #2811, @jongwooo)
* `describe` command now supports descriptions of policycrds and namespaces.
  (#2872, #2836, @Devaansh-Kumar)
* We've added the ability to filter resources using labels (through the `-l`
  flag) with both the `get` and `describe` commands. (#2892, #2915, #2934,
  @yeedove)
* Bug fix: Prevent panic when describing gatewayclasses with no description
  (#2894, @pmalek)
```

- [v0.1.0-rc2](#v010-rc2)
- [v0.1.0-rc1](#v010-rc1)

> # v1.1.0-rc1

Reviewer 6: @robscott Do you want to mention the bump to go 1.22 and k8s 1.30?

Author: Good question, that will depend on the resolution to https://github.com/kubernetes-sigs/gateway-api/pull/2988#issuecomment-2065320460, will leave this open until that path becomes more clear.

## Experimental Channel

### Session Persistence + BackendLBPolicy
Session Persistence is being introduced to Gateway API via a new policy
(BackendLBPolicy) for Service-level configuration and as fields within HTTPRoute and
GRPCRoute for Route-level configuration. The BackendLBPolicy and Route-level
APIs provide the same session persistence configuration, including session
timeouts, session name, session type, and cookie lifetime type.

Leading Contributors: @gcs278, @ginayeh

### Gateway Client Cert Verification
Gateways can now configure client cert verification for each Gateway Listener by introducing a new field
`frontendValidation` field within `tls`. This field supports configuring a list of CA Certificates that can be used as a trust anchor to validate the certificates presented by the client.

Leading Contributors: @arkodg

### BackendTLSPolicy
As part of a broader goal of making our TLS terminology more consistent
throughout the API, we've introduced some breaking changes to BackendTLSPolicy.
This has resulted in a new API version (v1alpha3) and will require any existing
users of this policy to uninstall the v1alpha2 version before installing this
newer version.

> 

Reviewer 6: (this is alot of detail, but if this is what you wanted, please add)
```suggestion
Any references to v1alpha2 BackendTLSPolicy fields will need to be updated.  Specific changes include: 
* the field TargetRef is changed to allow multiples: BackendTLSPolicySpec.TargetRef  becomes a slice named TargetRefs.
* the field TLS is changed to Validation, and the type name changes from BackendTLSPolicyConfig to  BackendTLSPolicyValidation.
* the field CACertRefs becomes CACertificateRefs in what is now BackendTLSPolicyValidation
* the field WellKnownCACerts becomes WellKnownCACertificates and the type name changes from WellKnownCACertType to WellKnownCACertificatesType.
* the constant WellKnownCACertSystem becomes WellKnownCACertificatesSystem
```

Leading Contributors: @candita

### Gateway Params
Gateways now feature a new field that allows references to
implementation-specific parameters, similar to GatewayClass.

Leading Contributors: @howardjohn

## Everything Else

### Validation Changes
- TLS Configuration is no longer required on Gateway Listeners to enable more
  flexible TLS configuration. (#2721, @robscott)

> ### Conformance

Reviewer 6: ```suggestion
### Conformance Tests
```

## Experimental Channel

### Session Persistence + BackendLBPolicy
Session Persistence is being introduced to Gateway API via a new policy
(BackendLBPolicy) for Service-level configuration and as fields within HTTPRoute and
GRPCRoute for Route-level configuration. The BackendLBPolicy and Route-level
APIs provide the same session persistence configuration, including session
timeouts, session name, session type, and cookie lifetime type.

Leading Contributors: @gcs278, @ginayeh

### Gateway Client Cert Verification
Gateways can now configure client cert verification for each Gateway Listener by introducing a new field
`frontendValidation` field within `tls`. This field supports configuring a list of CA Certificates that can be used as a trust anchor to validate the certificates presented by the client.

Leading Contributors: @arkodg

### BackendTLSPolicy
As part of a broader goal of making our TLS terminology more consistent
throughout the API, we've introduced some breaking changes to BackendTLSPolicy.
This has resulted in a new API version (v1alpha3) and will require any existing
users of this policy to uninstall the v1alpha2 version before installing this

> newer version.
> Any references to v1alpha2 BackendTLSPolicy fields will need to be updated.  Specific changes include: 
> * the field TargetRef is changed to allow multiples: BackendTLSPolicySpec.TargetRef  becomes a slice named TargetRefs.
> * the field TLS is changed to Validation, and the type name changes from BackendTLSPolicyConfig to  BackendTLSPolicyValidation.
> * the field CACertRefs becomes CACertificateRefs in what is now BackendTLSPolicyValidation
> * the field WellKnownCACerts becomes WellKnownCACertificates and the type name changes from WellKnownCACertType to WellKnownCACertificatesType.
> * the constant WellKnownCACertSystem becomes WellKnownCACertificatesSystem

Reviewer 6: ```suggestion
newer version.

Any references to v1alpha2 BackendTLSPolicy fields will need to be updated.  Specific changes include: 
- the field TargetRef is changed to allow multiples: BackendTLSPolicySpec.TargetRef  becomes a slice named TargetRefs.
- the field TLS is changed to Validation, and the type name changes from BackendTLSPolicyConfig to  BackendTLSPolicyValidation.
- the field CACertRefs becomes CACertificateRefs in what is now BackendTLSPolicyValidation
- the field WellKnownCACerts becomes WellKnownCACertificates and the type name changes from WellKnownCACertType to WellKnownCACertificatesType.
- the constant WellKnownCACertSystem becomes WellKnownCACertificatesSystem
```

Leading Contributors: @candita

### Gateway Params
Gateways now feature a new field that allows references to
implementation-specific parameters, similar to GatewayClass.

Leading Contributors: @howardjohn

## Everything Else

### Validation Changes
- TLS Configuration is no longer required on Gateway Listeners to enable more
  flexible TLS configuration. (#2721, @robscott)

### Conformance Tests

- Fixed GatewayWithAttachedRoutes conformance test to not check that the
  HTTPRoute status includes an "Accepted: False" condition because this is not
  required by the specification. (#2548, @frankbu)

### Cleanup

- The validating webhook has been removed. CEL validation is now built-in to
  CRDs and replaces the webhook. (#2595, @robscott)
- BackendTLSPolicy WellKnownCACerts field has been updated to
  implementation-specific support (#2741, @sunjayBhatia)
- Clarify policy attachment by two of the same policy types when using section
  names. (#2442, @maleck13)

> 

Reviewer 6: Just in case people wonder why links broke
```suggestion
- Remove v1alpha2 directory from docs: (#2965, @robscott)
```

Leading Contributors: @howardjohn, @keithmattix, @kflynn

### Conformance Profiles and Reports

The Conformance Reports API and the corresponding test suite have been graduated to GA. The Conformance report API has been expanded with the `mode` field (intended to specify the working mode of the implementation), and the `gatewayAPIChannel` (standard or experimental). The `gatewayAPIVersion` and `gatewayAPIChannel` are now filled in automatically by the suite machinery, along with a brief description of the testing outcome.
The Reports have been reorganized in a more structured way, and the implementations can now add information on how the tests have been run and provide reproduction steps.

Leading Contributors: @mlavacca, @shaneutt

> ### Port in ParentRefs

Reviewer 10: Should this say `Port in ParentRefs (GEP-957) Promoted to GA`, or something like that?

Reviewer 11: `### ParentRef Port field Graduates to GA` ?
`### New GA Field: parentRef.port` ?

There's no way to do this that isn't a bit awkward, but this is going to be a common thing that happens so maybe we should think about how we want to do this in general...

GRPCRoute has graduated to GA (v1) and is now part of the Standard Channel. If
you are already using the experimental version GRPCRoute, we recommend holding
off on upgrading to the standard channel version of GRPCRoute until the
controllers you're using have been updated to support GRPCRoute v1. Until then,
it is safe to upgrade to the experimental channel version of GRPCRoute in v1.1
that includes both v1alpha2 and v1 API versions.

Leading Contributor: @gnossen

### Service Mesh Support has Graduated to GA 🎉

The standard for using Gateway API for Mesh has formally graduated to GA (v1) and is now part of the Standard Channel.

Service mesh support in Gateway API allows service mesh users to use the same API to manage ingress traffic and mesh traffic, reusing the same policy and routing interfaces. In Gateway API v1.1, routes (such as HTTPRoute) can now have a `Service` as a `parentRef`, to control how traffic to specific services behave. For more information, read the [service mesh](https://gateway-api.sigs.k8s.io/mesh/) documentation or the see the list of [integrations](https://gateway-api.sigs.k8s.io/implementations/#service-mesh-implementation-status).

> Leading Contributors: @howardjohn, @keithmattix, @kflynn

Reviewer 11: ```suggestion
Leading Contributors: @howardjohn, @keithmattix, @mikemorris, @kflynn
```

Reviewer 11: (FTR I like the text that @howardjohn provided. 🙂)

Author: Thanks @kflynn! Do you mind updating your suggestion to be in alphabetical order by GitHub username just for the sake of consistency with the other lists?

Author: JK, realized I could change your suggest 🙃, hope this tweak is OK.
